persistence-modules/spring-data-cosmosdb/src/test/java/com/baeldung/spring/data/cosmosdb/AzurecosmodbApplicationManualTest.java

```diff
@@ -0,0 +1,34 @@
+package com.baeldung.spring.data.cosmosdb;
+
+import com.azure.data.cosmos.PartitionKey;
+import com.baeldung.spring.data.cosmosdb.entity.Product;
+import com.baeldung.spring.data.cosmosdb.repository.ProductRepository;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.util.Assert;
+
+import java.util.Optional;
+
+@SpringBootTest
+public class AzurecosmodbApplicationManualTest {
+
+    @Autowired
+    ProductRepository productRepository;
+
+    @Test
+    public void givenProductIsCreated_whenCallFindById_thenProductIsFound() {
+        Product product = new Product();
+        product.setProductid("1001");
+        product.setProductCategory("Shirt");
+        product.setPrice(110.0);
+        product.setProductName("Blue Shirt");
+
+        productRepository.save(product);
+        Optional<Product> retrievedProduct = productRepository.findById("1001", new PartitionKey("Shirt"));
+        Assert.notNull(retrievedProduct.get(), "Retrieved Product is Null");
```

Comment:
no.. this will produce a null pointer if it's null and the test is not doing the same thing. just do:
Optional<Product> retrievedProduct = productRepository.findById("1001", new PartitionKey("Shirt")).orElse(null); and then assert the variable